ccmlib/hcd/hcd_node.py

```diff
@@ -0,0 +1,237 @@
+import tempfile
+import yaml
+
+from distutils.version import LooseVersion
+from six.moves import urllib, xrange
+
+from ccmlib import common, node, repository
+from ccmlib.common import ArgumentError, rmdirs
+from ccmlib.node import Node
+
+
+
+HCD_ARCHIVE = "http://downloads.datastax.com/hcd/hcd-%s-bin.tar.gz"
+
+class HcdNode(Node):
+
+    """
+    Provides interactions to a HCD node.
+    """
+
+    @staticmethod
+    def get_version_from_build(install_dir=None, node_path=None, cassandra=False):
+        if install_dir is None and node_path is not None:
+            install_dir = node.get_install_dir_from_cluster_conf(node_path)
+        if install_dir is not None:
+            # Binary cassandra installs will have a 0.version.txt file
+            version_file = os.path.join(install_dir, '0.version.txt')
+            if os.path.exists(version_file):
+                with open(version_file) as f:
+                    return LooseVersion(f.read().strip())
+            # For DSE look for a dse*.jar and extract the version number
+            hcd_version = get_hcd_version(install_dir)
+            if (hcd_version is not None):
+                if cassandra:
+                    return get_hcd_cassandra_version(install_dir)
+                else:
+                    return LooseVersion(hcd_version)
+            # Source cassandra installs we can read from build.xml
+            return Node.get_version_from_build(install_dir, cassandra)
+        raise common.CCMError("Cannot find version")
+
+
+    def __init__(self, name, cluster, auto_bootstrap, thrift_interface, storage_interface, jmx_port, remote_debug_port, initial_token, save=True, binary_interface=None, byteman_port='0', environment_variables=None, derived_cassandra_version=None, **kwargs):
+        super(HcdNode, self).__init__(name, cluster, auto_bootstrap, thrift_interface, storage_interface, jmx_port, remote_debug_port, initial_token, save, binary_interface, byteman_port, environment_variables=environment_variables, derived_cassandra_version=derived_cassandra_version, **kwargs)
+
+    def get_install_cassandra_root(self):
+        return os.path.join(self.get_install_dir(), 'resources', 'cassandra')
+
+    def get_node_cassandra_root(self):
+        return os.path.join(self.get_path(), 'resources', 'cassandra')
+
+    def get_conf_dir(self):
+        """
+        Returns the path to the directory where Cassandra config are located
+        """
+        return os.path.join(self.get_path(), 'resources', 'cassandra', 'conf')
+
+    def get_tool(self, toolname):
+        return common.join_bin(os.path.join(self.get_install_dir(), 'resources', 'cassandra'), 'bin', toolname)
+
+    def get_tool_args(self, toolname):
+        return [common.join_bin(os.path.join(self.get_install_dir(), 'resources', 'cassandra'), 'bin', toolname)]
+
+    def get_env(self):
+        env = self.make_hcd_env(self.get_install_dir(), self.get_path())
+        # adjust JAVA_HOME to one supported by this dse_version
```

Comment:
```suggestion
        # adjust JAVA_HOME to one supported by this hcd_version
```

Reply:
I'm not 100% sure of this, but maybe this should read as `ds_version`, which seems to match the below file from the HCD tar binary download:
<img width="172" alt="image" src="https://github.com/user-attachments/assets/c6f2bdae-f9b9-40d1-8b83-d117e71fa874" />

Reply:
IIUC it goes to `get_version_from_build(install_dir)` above, i.e.
https://github.com/apache/cassandra-ccm/pull/784/files#diff-75cd9ea4fa90cbe59a44985da0b555af6856775cc8bbbe04720bac0b702c360dR47

i don't see anywhere that ds_version.txt is used 🤷